File: issuebook.py

bookname=input("enter the book name:")
rollno=input("enter the roll number:")
f=open("file1.txt",'r')#contains all the names of the book
p=open("file2.txt",'r')#contains the count of each book
d=open("file3.txt",'a')
list=f.read().split('\n')
#print(list)#splitting the list by the new line
list2=p.read().split('\n')
#print(list2)#splitting the list by the new line
while('' in list2):
    list2.remove('')#for the first execution it works fine but for the second execution  an empty string will be in the list inorder to remove we use this
#print(list2)

if bookname  in list :
    y=bookname#storing the book name in the variable y
    index=list.index(y)
    #print(index)
    x=(list2[index])#accesing the count of books from the list2 i.e the count of each group
    #print(x)
    if(int(x)>0):
        print("the book  can be issued")
        x=int(x)-1#if the book is issued then we need to decrease the count of book 
        print(" the book has been issued to \t" + rollno)
        d=open("file3.txt",'a')
        d.write("\nthe" +" "+bookname+" " + "has been issued to "+ rollno+"\n")#storing the data into the file
        print("The remaining books left is ",str(x))
        list2[index]=str(x)
        #print(list2)
        l=open("file2.txt",'w')
        for items in list2:
            l.write(items+"\n")#after the count has been decreased then changing the count in the list
        l.close()
        if(int(x)==0):
            print("the book stock has been completed")
    else:
        print("stock has been completed")
    
else :
    print("book is not available")

File: IssueBook.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Scanner;

public class IssueBook {
    public static boolean verify= false;
    public static Scanner sc = new Scanner(System.in);

    // String id=s.next();
    public static void write(String id, File fi) throws IOException {// creating a method to write the data to file
        FileWriter fw = new FileWriter(fi, true);
        fw.write(id);
        fw.close();
    }

    public static String[] readFiles(String file) {// converting the data of the file into array and storing
        try {
            File f = new File(file);
            Scanner s = new Scanner(f);
            int c = 0;
            int flag = 0;
            while (s.hasNextLine()) {
                c++;
                s.nextLine();
            }
            String[] arr = new String[c];
            Scanner s1 = new Scanner(f);
            for (int i = 0; i < arr.length; i++) {
                arr[i] = s1.nextLine();
                
                //System.out.println(arr[i]);
            }
            
            System.out.println("Enter the bookname:");
            String bname = sc.nextLine();
            
             //System.out.print(bname);
            for (int i = 0; i < arr.length; i++) {
                //System.out.println(arr[i]);
                if (bname.equalsIgnoreCase(arr[i])) {// making sure the username is not repeated
                    int index=0;
                    index=i;
                    verify=true;
                   // System.out.println("index is "+""+i);
                   /* while(i!=-1)
                    { 
                        int[] indices
                        int index1=-1;


                    }
                 */
                    flag = 1;
                    break;
                } else {
                    flag = 0;
                }
            }
            
            
            if (flag == 1) {
                int count=0;
                System.out.println("The book is available");
                
                     try {
                           File file1 = new File("books.txt");
                           FileWriter store = new FileWriter(file1,true);
                           PrintWriter pw = new PrintWriter(store);
                           pw.println(count+" "+bname+"is issued");
                           count++;
                           System.out.println(count+" "+"book has been issued");
                           pw.close();
                           
                           
                     }catch (IOException e) {
                           System.out.println("An error occurred.");
                            e.printStackTrace();
                    }
            }

              
             else {
                System.out.println("The book is not available");

            }
            return arr;
        }catch(

    Exception e)
    {
        return null;
    }
    }

    public static void main(String[] args) {
        String[] data = readFiles("bookname.txt");
    }
}
